modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1945,7 +1946,12 @@ def get_axis_lengths(partitions, axis):
 
         # If all frames are empty
         if len(non_empty_frames_idx) == 0:
-            return self._partitions, [o._partitions for o in other], joined_index
+            return (
+                self._partitions,
+                [o._partitions for o in other],
+                joined_index,
+                None,
```

Comment:
```suggestion
                [],
```
or 
```suggestion
                [0] * (len(other) + 1),
```
so we always return something consistent from the function